sql/sql_connect.cc

```diff
@@ -826,6 +826,21 @@ bool init_new_connection_handler_thread()
   return 0;
 }
 
+static bool check_require_secured_transport(THD *thd)
+{
+  Vio *vio= thd->net.vio;
+  if (opt_require_secure_transport)
+  {
+    enum enum_vio_type type= vio_type(vio);
+
+    return
+      (type != VIO_TYPE_SSL) &&
+      (type != VIO_TYPE_NAMEDPIPE);
+      (type != VIO_TYPE_SOCKET);
```

Comment:
@Chupsy, these should be separated by `;` rather than `&&`… probably should be caught by a compiler warning.